server/src/main/java/org/opensearch/index/engine/NRTReplicationNoOpEngine.java

```diff
@@ -0,0 +1,116 @@
+ * acks the primary with the expected value as the status quo. There is a plan to remove this or refactor this basis the
+ * primary-primary recovery with SegRep and Remote store for translog. If we were to remove this engine, we can programme
+ * the {@link org.opensearch.action.bulk.TransportShardBulkAction#dispatchedShardOperationOnReplica(BulkShardRequest,
+ * IndexShard, ActionListener)} method to complete the listener on a dummy response.
+ *
+ * @opensearch.internal
+ */
+public class NRTReplicationNoOpEngine extends AbstractNRTReplicationEngine {
+
+    private static final TranslogStats DEFAULT_STATS = new TranslogStats();
+
+    public NRTReplicationNoOpEngine(EngineConfig engineConfig) {
+        super(engineConfig);
+    }
+
+    @Override
+    protected TranslogManager createTranslogManager(String translogUUID, SetOnce<TranslogManager> translogManager) throws IOException {
+        return new NoOpTranslogManager(shardId, readLock, this::ensureOpen, null, new Translog.Snapshot() {
+            @Override
+            public void close() {}
+
+            @Override
+            public int totalOperations() {
+                return 0;
+            }
+
+            @Override
+            public Translog.Operation next() {
+                return null;
+            }
+        }) {
+            @Override
+            public TranslogStats getTranslogStats() {
+                return DEFAULT_STATS;
+            }
+        };
+    }
+
+    @Override
+    public long getLastSyncedGlobalCheckpoint() {
+        return -1;
+    }
+
+    /**
+     * This method is essentially no-op but does however updates in-memory local checkpoint tracker.
+     *
+     * @param index request for index
+     * @return index result.
+     */
+    @Override
+    public IndexResult index(Index index) throws IOException {
+        return super.index(index);
+    }
+
+    /**
+     * This method is essentially no-op but does however updates in-memory local checkpoint tracker.
+     *
+     * @param delete request for delete.
+     * @return delete result.
+     */
+    @Override
+    public DeleteResult delete(Delete delete) throws IOException {
+        return super.delete(delete);
+    }
+
+    /**
+     * This method is essentially no-op but does however updates in-memory local checkpoint tracker.
+     *
+     * @param noOp request for no-op.
+     * @return no-op result.
+     */
+    @Override
+    public NoOpResult noOp(NoOp noOp) throws IOException {
+        return super.noOp(noOp);
+    }
+
+    /**
+     * This method tracks the maximum sequence number of the request that has been given for indexing to this replica.
+     * Currently, the recovery process involves replaying translog operation on the replica by the primary. For recovery
+     * step to finish, and finalize step to kick in, this method should return expected value. Hence it has been overridden.
```

Comment:
Would `expected value` be equal to `maxSeqNo` always?

Reply:
so, what happens currently (in translog replay from primary during recovery) is that the applyIndexOperation is performed. This in the doc replication world would sync to translog before acking back and hence updating the persisted sequence number in local checkpoint tracker. AFAIK, this is expected. Since, we are freeing translog on replicas, we have to have this interim solution for recovery to work and when we tweak the recovery for no-op replication use case, the method might become unnecessary on replica's engine.